### DISCUSSION TITLE
GPU switching

### DISCUSSION THREAD

**Discussion by WILLYWZ** [seerge/g-helper#1449] [General]
I know it might sound weird, nut when you unplug the device ( so it run ob battery), software(s) that are using dGPU are "kicked Out". For example, Unity3D gives you a message that you need to restart because gpu is no longer available. I think the problem is that there are a small gap when switching from dGPU to iGPU. Armory crate, even on optimus, still keeps the software running on dGPU. Is there possible to make a whitelist on softwares that don't want it to switch immediately? Thanks. 
<img width="301" alt="image" src="https://github.com/seerge/g-helper/assets/49498287/51ccbd41-93ac-4063-b645-e166a747f135">

**Comment by WILLYWZ** [comment]
i don't know if this is an issue while gaming, and i will look into that.

**Reply by WILLYWZ** [reply]
and i will look into unity settings this week, and if i found a way to set GPU, i will close this discussion.

**Comment by seerge** [comment]
@WILLYWZ hello. 

If you have Optimized GPU mode it (by definition) will turn off dGPU completely when you unplug your laptop.

But there is already a special check for GPU load, and if it's higher than 10% - it won't do that and ask you first.
https://github.com/seerge/g-helper/blob/main/app/Gpu/GPUModeControl.cs#L232

I assume that nothing is actually using your dGPU actively on the moment of unplugging then.

If you want dGPU to be always available (and controlled by optimus) just keep Standard mode

**Comment by WILLYWZ** [comment]
Thanks. I guess it is using it, but not yet to the 10 percent threshold.

**Comment by frozen-pizza** [comment]
hi there, i'm also experiencing this issue. However, this occur to me while connecting an external monitor. It seems like optimized mode does not support external display. Whenever my laptop is connected to an external monitor while starting up on battery, it will freeze up and must be forced to shut down. Not a big issue, but hopefully it can be fixed.

**Reply by seerge** [reply]
@frozen-pizza "Optimized" mode is just an automatic switch between Standard (dGPU ON) when plugged and Eco (dGPU OFF) when on battery, nothing more. There is nothing to "fix" here.

If you start your computer with dGPU disabled and monitor connected, make sure that you connect it to the correct usb-c port. Rest is up to Windows and GPU drivers (you may try to reinstall them using DDU)

**Reply by frozen-pizza** [reply]
I see.. I connected my display to the HDMI port. But I will try to reinstall my drivers.

**Reply by seerge** [reply]
HDMI port most probably routed to dGPU (at least it's like that on all laptops that I personally had like G14 and G16), so your monitor won't work when it's off :) Just set mode to Standard then and you will be fine

**Reply by frozen-pizza** [reply]
exactly. probably will stay on standard mode for the time being.

**Reply by Vindo11ar** [reply]
Yeah I have got something similar going on aswell, where I cannot use my external monitor with Eco mode (I have display port to USB-C into the left USB C on G14). When I switch to eco mode, monitor will black and say no signal. So if i want to use external monitor I have to keep it in Standard mode.
 
I just wanted to save a bit of battery, but ext monitor drains like crazy when just browsing. Have to charge every hour.

Any way to allow Eco mode and Ext monitor to use dGPU and save battery?

Thanks
